src/App.jsx

```diff
@@ -1,10 +1,106 @@
+            setCountryName("");
+        } catch (e) {
+            console.error(e)
+            setError(`${countryName} bestaat niet. Probeer het opnieuw.`)
+        }
+    }
+
     return (
         <>
-            Maak je applicatie hier!
+            <main className="page-container">
+                <section className="all-countries-section">
+                <span className="header-image-wrapper">
+                    <img className="header-image" src={worldmap} alt="world-map"/>
+                </span>
+                        <button className="all-countries-button" type="button" onClick={fetchAllCountries}>Zoek alle landen</button>
+                    <ul className="countries-container">
+                        {countries && countries.map((country) => (
+                            <li key={country.cca2}>
+                                <section className="all-country-cards">
+                                    <div className="country-information">
+                                        <img className="country-flag-images"
+                                             src={country.flags.png}
+                                             alt={country.name.common}
+                                        />
+                                        <div className="country-title">
+                                            <h1 className={regionColorNames(country.region)}>{country.name.common}</h1>
+                                        </div>
+                                    </div>
+                                    <p className="country-population">Has a population of {country.population}</p>
+                                </section>
+                            </li>
+                        ))}
+                    </ul>
+                </section>
+                <section className="individual-country-section">
+                    <h1>Search country information</h1>
+                    <span className="globe-image-wrapper">
+                        <img className="globe-image" src={globe} alt="globe"/>
+                    </span>
+                    <div className="country-search-bar">
+                        <input
+                            type="text"
+                            className="single-country-input"
+                            placeholder="Bijvoorbeeld Nederland of Peru"
+                            value={countryName}
+                            onChange={(event) => setCountryName(event.target.value)}
+                        />
+                        <button
+                            type="button"
+                            className="single-country-search-button"
+                            onClick={fetchCountry}
+                        >
+                                ZOEK
+                        </button>
+                        {error && <span id="error-message">{error}</span>}
+                    </div>
+                    {Object.keys(countryData).length > 0 &&
```

Comment:
```suggestion
                    {countryData &&
```
(Mits de default waarde `undefined` is)